mlir/lib/Dialect/Linalg/Transforms/DecomposeGenericByUnfoldingPermutation.cpp

```diff
@@ -125,7 +124,7 @@ computeTransposeBroadcast(AffineMap &map) {
     // refering to d3, d4. Therefore, re-base the transpose dimensions so
     // that they start from d0.
     permutation.resize(minorSize);
-    std::map<int64_t, int64_t> minorMap;
+    llvm::DenseMap<int64_t, int64_t> minorMap;
```

Comment:
Seems to me that this is all a map of positions right? So why can't we use a `SmallVector<int>` here?

Reply:
I don’t quite follow: we have key-value pair, the natural translation when you have keys that are small integers is an array of values, I don’t see where the pair would come from (the position in the array is the key)

Reply:
That makes sense. I wonder in that case if the original intention to maintain a map is for cases such as: 
`[1,2,4]`
.. where a dimension isn't present and you wouldn't be able to rely on the index being equivalent to the key.

Reply:
Possibly, but we never iterate the map, so having unused entries does not matter I believe.